src/app/shared/components/pagination/pagination.component.ts

```diff
@@ -0,0 +1,46 @@
+import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
+
+@Component({
+  selector: 'app-pagination[tableName]',
+  templateUrl: './pagination.component.html',
+  styleUrls: ['./pagination.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PaginationComponent {
+  @Input() tableName!: string;
+  @Input() itemsPerPage: number = 5;
+  @Input() numberOfItems: number = 0;
+  @Input() currentPage: number = 1;
+
+  constructor() {}
+
+  pageQuery(page: number) {
+    return { [`${this.tableName}-page`]: page };
+  }
+  nextPage() {
+    return this.pageQuery(this.currentPage + 1);
+  }
+  prevPage() {
+    return this.pageQuery(this.currentPage - 1);
+  }
+
+  trackByFn(index: number, page: number) {
+    return page || index;
+  }
+
+  get pages() {
+    return Array(this.numberOfPages)
+      .fill('')
+      .map((x, i) => i + 1);
+  }
+
+  get numberOfPages() {
+    return Math.ceil((this.numberOfItems ?? 0) / this.itemsPerPage);
```

Comment:
`numberOfItems` defaults to 0 if it's not an input and is always defined 
```suggestion
    return Math.ceil(this.numberOfItems / this.itemsPerPage);
```